zha/zigbee/device.py

```diff
@@ -1087,3 +1102,167 @@ def log(self, level: int, msg: str, *args: Any, **kwargs: Any) -> None:
         msg = f"[%s](%s): {msg}"
         args = (self.nwk, self.model) + args
         _LOGGER.log(level, msg, *args, **kwargs)
+
+    def get_diagnostics_json(self):
+        """Get ZHA device information."""
+
+        info: dict[str, Any] = {}
+        info["ieee"] = str(self.ieee)
+        info["nwk"] = self.nwk
+        info["manufacturer"] = self.manufacturer
+        info["model"] = self.model
+        info["name"] = self.name
+        info["quirk_applied"] = self.quirk_applied
+        info["quirk_class"] = self.quirk_class
+        info["quirk_id"] = self.quirk_id
+        info["manufacturer_code"] = self.manufacturer_code
+        info["power_source"] = self.power_source
+        info["lqi"] = self.lqi
+        info["rssi"] = self.rssi
+        info["last_seen"] = self.device._last_seen.isoformat()
+        info["available"] = self.available
+        info["device_type"] = self.device_type
+        info["active_coordinator"] = self.is_active_coordinator
+
+        node_desc = self.device.node_desc
+        info["node_descriptor"] = {
```

Comment:
how come this is handled field by field here but `as_dict` is used below when processing `original_signature` ?

Reply:
`as_dict` doesn't convert enum member names and instead leaves them as integers. It's not very readable.

Reply:
Ah right. Forgot about that.

Reply:
Could we use the encoder to do the conversion?

Reply:
We should have JSON-serializable version of all of these (device info, original signature, entity state, and topology info) once the Pydantic PRs start coming in. I think this should really clean up the code in `device.py` and move it into `models.py`

Reply:
#soon